src/main/java/edu/cornell/kfs/pdp/service/impl/CuPaymentDetailServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package edu.cornell.kfs.pdp.service.impl;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.core.api.criteria.CriteriaLookupService;
+import org.kuali.kfs.core.api.criteria.GenericQueryResults;
+import org.kuali.kfs.core.api.criteria.Predicate;
+import org.kuali.kfs.core.api.criteria.PredicateFactory;
+import org.kuali.kfs.core.api.criteria.QueryByCriteria;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.pdp.PdpPropertyConstants;
+import org.kuali.kfs.pdp.businessobject.PaymentDetail;
+import org.kuali.kfs.pdp.service.impl.PaymentDetailServiceImpl;
+import org.kuali.kfs.sys.DynamicCollectionComparator;
+
+import edu.cornell.kfs.pdp.service.CuPaymentDetailService;
+
+public class CuPaymentDetailServiceImpl extends PaymentDetailServiceImpl implements CuPaymentDetailService {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private CriteriaLookupService criteriaLookupService;
+
+    /*
+     * Copied the superclass's 4-arg version of this method and added handling of the Process Immediate flag.
+     * Also adjusted the query to use the CriteriaLookupService, since the Process Immediate flag could
+     * potentially be null.
+     */
+    @Override
+    public Iterator<PaymentDetail> getByDisbursementNumber(Integer disbursementNumber, Integer processId,
+            String disbursementType, String bankCode, boolean processImmediate) {
+        LOG.debug("getByDisbursementNumber() started");
+
+        QueryByCriteria criteria = QueryByCriteria.Builder.fromPredicates(
+                PredicateFactory.equal(PdpPropertyConstants.PaymentDetail.PAYMENT_DISBURSEMENT_NUMBER,
+                        disbursementNumber),
+                PredicateFactory.equal(PdpPropertyConstants.PaymentDetail.PAYMENT_GROUP + "." +
+                        PdpPropertyConstants.PaymentGroup.PAYMENT_GROUP_PROCESS_ID, processId),
+                PredicateFactory.equal(PdpPropertyConstants.PaymentDetail.PAYMENT_GROUP + "." +
+                        PdpPropertyConstants.PaymentGroup.PAYMENT_GROUP_DISBURSEMENT_TYPE_CODE, disbursementType),
+                PredicateFactory.equal(PdpPropertyConstants.PaymentDetail.PAYMENT_GROUP + "." +
+                        PdpPropertyConstants.PaymentGroup.PAYMENT_GROUP_BANK_CODE, bankCode),
+                createProcessImmediateCondition(processImmediate));
+        GenericQueryResults<PaymentDetail> results = criteriaLookupService.lookup(PaymentDetail.class, criteria);
```

Comment:
Base code is using businessObjectService to look up payment details, and Map<String, Object> fieldValues for the search criteria.  Would it be better to use base code style to make bringing in kualicp patches easier?

Reply:
I wasn't sure whether we needed to handle the possibility of the `processImmediate` flag being null on the Payment Group. I haven't found any examples where this flag has been set to null in the database, though, so it might be safe to treat it as non-null, in which case I could switch this back to using the Business Object Service.

Thoughts?

Reply:
Given my comments above, I'm switching this back to using the Business Object Service, since it looks like the process-immediate flag is not being left as null (and since certain PaymentGroup-processing areas of KFS code already assume the flag is non-null). If you think the null-checking solution should be used instead, please let me know.